This is about python exceptions
